app/helpers/barnardos/action_view/form_helpers.rb

```diff
@@ -3,23 +3,29 @@ module ActionView
     module FormHelpers
       ##
       # <!-- Example HTML output for labelled_text_field_tag :participant_name -->
-      # <div class="textfield " id="participant_name-wrapper">
-      #   <label class="textfield__label textfield__label--bold" for="participant_name">
+      # <div class="textfield js-textfield" id="participant_name-wrapper">
+      #   <label class="textfield__label" for="participant_name">
       #     What is the name of the research participant?
       #     <span class="textfield__hint">Full name</span>
       #   </label>
       #   <input id="participant_name" class="textfield__input" name="participant_name" type="text">
       # </div>
-      def labelled_text_field_tag(name, label, value: nil ,
+      def labelled_text_field_tag(name, label, value: nil , error: nil,
                                   text_options: {}, label_options: {})
-        content_tag :div, class: 'textfield', id: "#{name}-wrapper" do
+        className = 'textfield js-textfield'
+        if error
+          className += ' has-error'
+        end
+
+        content_tag :div, class: className, id: "#{name}-wrapper" do
```

Comment:
If you find it succinterer, you can use the post-script `if` form like you did on L28 to do something like:

```ruby
        content_tag :div, class: "textfield js-textfield #{' has-error' if error}", id: "#{name}-wrapper" do
```

That'd avoid the garbage collection for the `className` variable and the cost of the extra string concat. Rails views have a reputation for being slow, so whatever we can squeeze from helpers that we use multiple times helps.